### PR TITLE
Improve test coverage for XmlSerializer

### PR DESCRIPTION
**Pull request by khdang**

### CODE REVIEW

**Comment by khdang** [comment]
cc: @zhenlan @SGuyGe @Jinhuafei @shmao

File: src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs [hunk 1]
```diff
@@ -1130,6 +1170,174 @@ public static void Xml_TypeWithNonParameterlessConstructor()
...
+        System.Buffer.BlockCopy(str.ToCharArray(), 0, bytes, 0, bytes.Length);
+        return bytes;
+    }
+
+    private static string GetString(byte[] bytes)
+    {
+        char[] chars = new char[bytes.Length / sizeof(char)];
+        System.Buffer.BlockCopy(bytes, 0, chars, 0, bytes.Length);
+        return new string(chars);
+    }
```
Comment (stephentoub) [inline src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs]: Could you just use Encoding.Unicode.GetString/GetBytes, or something like that?

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 2]
```diff
@@ -2120,12 +2131,10 @@ public TypeWithNonParameterlessConstructor(string value)
     }
 }
 
-[DataContract]
-public class AppEnvironment
```
Comment (stephentoub) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: All of the builds failed due to:
```DataContractJsonSerializer.cs(1302,23): error CS0246: The type or namespace name 'AppEnvironment' could not be found (are you missing a using directive or an assembly reference?)```

File: src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs (hunk 1 above)
Comment (khdang) [inline src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs]: Yes I've updated to use that instead.

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs (hunk 2 above)
Comment (khdang) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: My bad this was due to an incorrect merge.

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**